modules/catalog/src/main/scala/lucuma/catalog/BrightnessConstraints.scala

```diff
@@ -27,19 +29,30 @@ case class BrightnessConstraints(
   faintnessConstraint:  FaintnessConstraint,
   saturationConstraint: Option[SaturationConstraint]
 ) derives Eq {
-  def contains(band: Band, brightness: BigDecimal): Boolean =
+  def contains(band: Band, brightness: BrightnessValue): Boolean =
     searchBands.bands.contains(band) &&
       faintnessConstraint.brightness >= brightness &&
       saturationConstraint.forall(_.brightness <= brightness)
 
   def ∪(that: BrightnessConstraints): BrightnessConstraints =
     BrightnessConstraints(
       searchBands ∪ that.searchBands,
-      FaintnessConstraint(faintnessConstraint.brightness.max(that.faintnessConstraint.brightness)),
+      FaintnessConstraint(
+        BrightnessValue
+          .from(
+            faintnessConstraint.brightness.value.value
```

Comment:
I think you should be able to `math.max(brightnessValue1, brightnessValue2)`. You may need an additional `import cats.Order.given`.

Reply:
yes, it is much nicer that way